cmd/flowlogs-pipeline/main.go

```diff
@@ -177,7 +179,15 @@ func run() {
 	health.NewHealthServer(mainPipeline)
 
 	// Starts the flows pipeline
-	mainPipeline.Run()
+	stopper := make(chan os.Signal, 1)
+	go func() {
+		mainPipeline.Run()
+		close(stopper)
+	}()
+
+	// Subscribe to signals for terminating the program.
+	signal.Notify(stopper, os.Interrupt, syscall.SIGTERM)
```

Comment:
This has to be integrated to work with exiting all the other threads that exit when SIGINT or SIGTERM are received. See SetupElegantExit() in flowlogs-pipeline/pkg/pipeline/utils/exit.go